File: milestone 2/python_modules/module03/ex0/ft_command_quest.cpp
#include <iostream>
#include <string>
using namespace std;

int main(int argc, char **argv)
{
    string program_name = argv[0];
    int args_received = argc - 1;
    int i = 1;

    cout << "=== Command Quest === \n\n";
    if (argc == 1)
    {
        cout << "No arguments provided!\n";
        cout << "Program name: " << program_name << '\n';
        cout << "Total arguments: " << argc << endl;
    }
    else if (argc > 1)
    {
        cout << "Program name: " << program_name << '\n';
        cout << "Arguments received: " << args_received << '\n';
        while (i < argc)
        {
            cout << "Argument " << i << ": " << argv[i] << '\n';
            i++;
        }
        cout << "Total arguments: " << argc << '\n';
    }
}

File: milestone 2/python_modules/module03/ex0/ft_command_quest.py
import sys

if __name__ == "__main__":
    print("=== Command Quest ===")

    if len(sys.argv) == 1:
        print("No arguments provided!")
        print(f"Program name: {sys.argv[0]}")
        print(f"Total arguments: {len(sys.argv)}")

    elif len(sys.argv) > 1:
        print(f"Program name: {sys.argv[0]}")
        print(f"Arguments received: {len(sys.argv)}")

        i = 1
        for args in sys.argv[1:]:
            print(f"Argument {i}: {args}")
            i += 1

        print(f"Total arguments: {len(sys.argv)}")
        